File: frontend/src/api/Comment.js
import { api, apiWithCookie } from "./axios";

export async function createComment(postId, contents, authorId) {
  try {
    const res = await apiWithCookie.post("/comments", {
      post_id: postId,
      user_id: authorId,
      contents,
    });

    return res.data;
  } catch (err) {
    console.error("게시글 작성 실패:", err);
    throw err;
  }
}

export async function getCommentsByPostId(postId) {
  try {
    const res = await api.get(`/comments/${postId}`);
    return res.data;
  } catch (err) {
    console.error("게시글 상세 불러오기 실패:", err);
    throw err;
  }
}

export async function updateComment(postId, contents) {
  try {
    const res = await apiWithCookie.put(`/comments/${postId}`, {
      contents,
    });
    return res.data;
  } catch (err) {
    console.error("게시글 수정 실패:", err);
    throw err;
  }
}
export async function deleteComment(postId) {
  console.log(postId);
  try {
    const res = await apiWithCookie.delete(`/comments/${postId}`);
    return res.data;
  } catch (err) {
    console.error("게시글 삭제 실패:", err);
    throw err;
  }
}

File: backend/controllers/comment.py
from fastapi import status, Body, Path
from fastapi.responses import JSONResponse

from services import comment

# 댓글 생성
async def create_comment(body: dict = Body(...)):
    try:
        post_id = comment.create_comment(
            post_id=body.get("post_id"),
            user_id=body.get("user_id"),
            comments=body.get("contents"),
        )
        return JSONResponse(
            {"message": "댓글 생성 완료", 
             "post_id": post_id}, status_code=status.HTTP_201_CREATED)
    except Exception as e:
        return JSONResponse(
            {"message": "댓글 생성 실패", "error":str(e)}, 
            status_code=status.HTTP_400_BAD_REQUEST)
        

# 게시글 상세 조회
async def get_comment_by_post_id(post_id: int = Path(...)):
    try:
        post = comment.get_comment_by_post_id(post_id)
        return JSONResponse(
            {
                "message":"댓글 조회 성공",
                "data": post
            }, 
            status_code=status.HTTP_200_OK)
    except Exception as e:
        return JSONResponse(
            {"message": "댓글 조회 실패", "error":str(e)},
            status_code=status.HTTP_404_NOT_FOUND)

# 게시글 수정
async def update_comment_by_id(
    comment_id: int = Path(...), 
    body: dict = Body(...)
):
    try:
        comments = body.get("contents")
        print(comments)
        updated = comment.update_comment_by_id(comment_id, comments)

        if not updated:
            return JSONResponse(
                {"message": "댓글 없음"}, 
                status_code=status.HTTP_404_NOT_FOUND
            )

        return JSONResponse(
            {"message": "댓글 수정 성공"},
            status_code=status.HTTP_200_OK
        )
    except Exception as e:
        return JSONResponse(
            {"message": "댓글 수정 실패", "error":str(e)},
            status_code=status.HTTP_400_BAD_REQUEST
        )

async def delete_comment_by_id(comment_id: int = Path(...)):
    try:
        deleted = comment.delete_comment_by_id(comment_id)
        if not deleted:
            return JSONResponse(
                {"message": "댓글 없음"}, 
                status_code=status.HTTP_404_NOT_FOUND)
        return JSONResponse(
            {"message": "댓글 삭제 완료"}, 
            status_code=status.HTTP_200_OK)
    except Exception as e:
        return JSONResponse(
            {"message": "댓글 삭제 실패", "error":str(e)},
            status_code=status.HTTP_400_BAD_REQUEST)
        